lib/audited/auditor.rb

```diff
@@ -246,6 +248,14 @@ def auditing_enabled
       def auditing_enabled=(val)
         self.class.auditing_enabled = val
       end
+
+      def skip_writing?
+        if Audited.only_authenticated_user
+          ::Audited.store[:current_controller] ? ::Audited.store[:current_controller].send(Audited.current_user_method).blank? : nil
```

Comment:
1. Could you use `store[:current_user]` instead? If the controller doesn't respond to the current user method then this would cause an error - the `:current_user` contains a lambda from Sweeper that properly checks this, you can re-use it for free. See `Audited::Audit#set_audit_user` for how to call it.
1. It should probably also `return false` if `store[:audited_user]` is set, because someone could call `Audited.as_user` even if the controller doesn't set the user.